modules/claims_api/app/services/claims_api/dependent_claimant_poa_assignment_service.rb

```diff
@@ -0,0 +1,169 @@
+        claimant_ssn: @claimant_ssn,
+        power_of_attorney: @poa_code,
+        benefit_claim_type: benefit_claim_type(claim_details[:pgm_type_cd]),
+        old_end_product_code: claim_details[:cp_claim_end_prdct_type_cd],
+        new_end_product_label: claim_details[:bnft_claim_type_cd],
+        old_date_of_claim: claim_rcvd_dt,
+        allow_poa_access: @allow_poa_access,
+        allow_poa_cadd: @allow_poa_cadd
+      }
+    end
+
+    def assign_poa_to_dependent_via_update_benefit_claim?
+      first_open_claim = dependent_claims.find do |claim|
+        claim[:phase_type] != 'Complete' && claim[:ptcpnt_vet_id] == @veteran_participant_id
+      end
+      first_open_claim_details = claim_details(first_open_claim[:benefit_claim_id])
+
+      benefit_claim_update_input = build_benefit_claim_update_input(claim_details: first_open_claim_details)
+
+      result = benefit_claim_service.update_benefit_claim(benefit_claim_update_input)
+
+      if result[:return][:return_message] == 'Update to Corporate was successful'
+        log(detail: 'POA assigned to dependent.')
+
+        return true
+      end
+
+      false
+    end
+
+    def dependent_claims
+      local_bgs = ClaimsApi::LocalBGS.new(external_uid: @dependent_participant_id,
+                                          external_key: @dependent_participant_id)
+      res = local_bgs.find_benefit_claims_status_by_ptcpnt_id(@dependent_participant_id)
+
+      return res&.dig(:benefit_claims_dto, :benefit_claim) if res&.dig(:benefit_claims_dto, :benefit_claim).present?
+
+      log(level: :error, detail: 'Dependent claims not found in BGS')
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    def benefit_claim_web_service
+      ClaimsApi::BenefitClaimWebService.new(external_uid: @dependent_participant_id,
+                                            external_key: @dependent_participant_id)
+    end
+
+    def benefit_claim_service
+      ClaimsApi::BenefitClaimService.new(external_uid: @dependent_participant_id,
+                                         external_key: @dependent_participant_id)
+    end
+
+    def claim_details(claim_id)
+      res = benefit_claim_web_service.find_bnft_claim(claim_id:)
+
+      return res&.dig(:bnft_claim_dto) if res&.dig(:bnft_claim_dto).present?
+
+      log(level: :error, detail: 'Claim details not found in BGS', claim_id:)
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    def poa_participant_id
+      poa_ptcpnt = FindPOAsService.new.response.find { |combo| combo[:legacy_poa_cd] == @poa_code }
+
+      return poa_ptcpnt&.dig(:ptcpnt_id) if poa_ptcpnt&.dig(:ptcpnt_id).present?
+
+      log(level: :error, detail: 'POA code/participant ID combo not found in BGS')
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    def manage_ptcpnt_rlnshp_poa_success?(response)
+      response.is_a?(Hash) && response.dig(:comp_id, :ptcpnt_rlnshp_type_nm) == 'Power of Attorney For'
```

Comment:
Brittle check. Any better suggestions?

Reply:
You could add some error handling that would alert you if the structure changed. 

Another idea is something like this:
```ruby
response.is_a?(Hash) && response.dig(:comp_id, :ptcpnt_rlnshp_type_nm)&.casecmp?('Power of Attorney For')
```
chatgpt suggested that, so definitely double check it's doing what you want 😃

Reply:
I thought about normalizing the strings, but I think failing loudly is our best bet at fixing the issue. I'm not sure trying to anticipate changing punctuation or cases is worthwhile 🤷